### ISSUE TITLE
Contacts not sorted correctly 

### ISSUE THREAD

**Issue by McTonderski** [signalapp/Signal-Android#10409]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Language specific letters are last in contact sorted list

### Steps to reproduce
- have contact with i.e. polish letter 

If for example name Is "Łukasz" it should be listed after contacts starting with "L" not at the end

### Device info

**Device:** KB2003
**Android version:** 11
**Signal version:** 5.0.8

**Comment by bkqtnte10** [comment]
Sorting related issue: #10384

**Comment by Caerbannog** [comment]
No, PR #10202 will not fix this issue, it is only concerned with finding search results.
Example result of the PR : searching for "e" will show "Mélissa" and "Éric". But "Éric" will still be sorted last, at "É" (I have tested this).

The two issues are related though.

**Comment by Caerbannog** [comment]
Yes, this issue is still relevant and irritating for lots of languages others than English.
Issue #10635 is a duplicate.

There are a few connected problems in Signal caused by _glyph similary_. *Contact sorting* is related but different from:
- Contact search (addressed in #10202 and subsequent improvements).
- Group name search
- Message content search

What's blocking is a design decision by the Signal team for how to solve these problems without duplicating code.
